app/src/main/java/com/architectcoders/aacboard/model/PictogramsRepository.kt

```diff
@@ -1,12 +1,24 @@
 package com.architectcoders.aacboard.model
 
+import com.architectcoders.aacboard.App
+import com.architectcoders.aacboard.model.database.DashboardEntityWithCellEntities
+import com.architectcoders.aacboard.model.datasource.DashboardLocalDataSource
+import kotlinx.coroutines.flow.Flow
 import java.util.*
 
-class PictogramsRepository {
+class PictogramsRepository(application: App) {
+
+    private val localDataSource = DashboardLocalDataSource(application.db.dashboardDao())
+
+    val dashboards= localDataSource.dashboards
 
     suspend fun searchPictograms(searchString: String): List<DomainPictogram> {
         val locale = Locale.getDefault().getLanguage()
         val arasaacPictograms = RemoteConnection.service.searchPictos(locale, searchString)
         return arasaacPictograms.map { it.toDomainPictogram() }
     }
+
+    fun getDashBoardWithCells(id: Int): Flow<DashboardEntityWithCellEntities> =
+        localDataSource.getDashBoardWithCells(id)
+
 }
```

Comment:
should we separate repositories? One DashboardRepository and one PictogramsRepository (to fetch arasac pictos)?

Reply:
It makes sense from a clean architecture perspective, but it may be overkill to have two repositories in 
 this simple app (PictogramsRepository will only have one method to call Arasaac API with a search query). 
Let us decide in the next call.